kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ConsumerOffsetsDaoImpl.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.controllers;
+
+import com.google.auto.value.AutoValue;
+import io.confluent.kafkarest.common.KafkaFutures;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.ConsumerGroupDescription;
+import org.apache.kafka.clients.admin.ListConsumerGroupOffsetsOptions;
+import org.apache.kafka.clients.admin.ListOffsetsOptions;
+import org.apache.kafka.clients.admin.ListOffsetsResult.ListOffsetsResultInfo;
+import org.apache.kafka.clients.admin.MemberDescription;
+import org.apache.kafka.clients.admin.OffsetSpec;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.ConsumerGroupState;
+import org.apache.kafka.common.IsolationLevel;
+import org.apache.kafka.common.TopicPartition;
+
+final class ConsumerOffsetsDaoImpl implements ConsumerOffsetsDao {
+
+  private final Admin kafkaAdminClient;
+  @Inject
+  ConsumerOffsetsDaoImpl(
+      Admin kafkaAdminClient
+  ) {
+    this.kafkaAdminClient = kafkaAdminClient;
+  }
+
+  @AutoValue
+  abstract static class MemberId {
+
+    abstract String getConsumerId();
+
+    abstract String getClientId();
+
+    abstract Optional<String> getInstanceId();
+
+    public static Builder builder() {
+      return new AutoValue_ConsumerOffsetsDaoImpl_MemberId.Builder();
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
+
+      abstract Builder setConsumerId(String consumerId);
+
+      abstract Builder setClientId(String clientId);
+
+      abstract Builder setInstanceId(Optional<String> instanceId);
+
+      abstract MemberId build();
+    }
+  }
+
+  @Override
+  public CompletableFuture<Map<TopicPartition, MemberId>> getMemberIds(
```

Comment:
I don't understand why does this method need to return a future. Nothing inside of it returns a future. The future you return is always complete.